File: Travel Package Management Java/src/com/example/travel/StandardPassenger.java
package com.example.travel;

public class StandardPassenger extends Passenger {
    public StandardPassenger(String name, int passengerNumber, double balance) {
        super(name, passengerNumber, balance);
    }

    @Override
    public boolean signUpForActivity(Destination destination, Activity activity) {
        double cost = activity.getCost();
        if (getBalance() >= cost && !activity.isSignedUp(this)) {
            if (activity.signUp(this)) {
                deductBalance(cost);
                return true;
            }
        }
        return false;
    }
}

File: Travel Package management C++/StandardPassenger.cpp
#include "StandardPassenger.h"

using namespace std;

bool StandardPassenger::signUpForActivity(shared_ptr<Destination> destination, shared_ptr<Activity> activity){
    double cost = activity->getCost();
    if(this->getBalance() >= cost && !activity->isSignedUp(this->shared_from_this())){
        if(activity->signUp(this->shared_from_this())){
            this->deductBalance(cost);
            return true;
        }
    }
    return false;
}
